core/module/used_refinements_spec.rb

```diff
@@ -0,0 +1,36 @@
+require_relative '../../spec_helper'
+
+describe "Module.used_refinements" do
+  ruby_version_is "3.2" do
+    it "returns list of all refinements imported (or activated/active) in the current scope" do
+      refinement_int = nil
+      refinement_str = nil
+      ScratchPad.record []
+
+      m1 = Module.new do
+        refine Integer do
+          refinement_int = self
+        end
+      end
+
+      m2 = Module.new do
+        refine String do
+          refinement_str = self
+        end
+      end
+
+      Module.new do
+        using m1
+        using m2
+
+        Module.used_refinements.each { |r| ScratchPad << r }
+      end
+
+      ScratchPad.recorded.should == [refinement_int, refinement_str]
+    end
+
+    it "returns empty array if does not have any refinements imported" do
+      Module.used_refinements.should == []
+    end
```

Comment:
**minor:** I suppose `Module.used_refinements` should be called in some module to not depend on test infrastructure. Now it checks a class where DSL methods are defined and executed.